sqlmesh/dbt/loader.py

```diff
@@ -67,10 +68,19 @@ def sqlmesh_config(
     if not issubclass(loader, DbtLoader):
         raise ConfigError("The loader must be a DbtLoader.")
 
+    if context.profile_name is None:
+        # Note: Profile.load() mutates `context` and will have already raised an exception if profile_name is not set,
+        # but mypy doesnt know this because the field is defined as t.Optional[str]
+        raise ConfigError(f"profile name must be set")
+
     return Config(
         loader=loader,
         model_defaults=model_defaults,
         variables=variables or {},
+        dbt_config_info=DbtConfigInfo(
+            profile_name=dbt_profile_name or context.profile_name,
```

Comment:
If we check `context.profile_name` above why not use it always since it's the source of truth?